foreman/data_refinery_foreman/surveyor/array_express.py

```diff
@@ -16,68 +17,111 @@
+            experiment["platform_accession_code"] = "UNKNOWN"
+        elif len(parsed_json["arraydesign"]) > 1:
+            experiment["platform_accession_code"] = "UNKNOWN"
+        else:
+            experiment["platform_accession_code"] = \
+                parsed_json["arraydesign"][0]["accession"]
+
+        experiment["release_date"] = parsed_json["releasedate"]
+
+        if "lastupdatedate" in parsed_json:
+            experiment["last_update_date"] = parsed_json["lastupdatedate"]
+        else:
+            experiment["last_update_date"] = parsed_json["releasedate"]
+
+        return experiment
+
+    def survey(self):
+        experiment_accession_code = (
+            SurveyJobKeyValue
+            .objects
+            .get(survey_job_id=self.survey_job.id,
+                 key__exact="experiment_accession_code")
+            .value
+        )
+
+        logger.info("Surveying experiment with accession code: %s.", experiment_accession_code)
+
+        experiment = self.get_experiment_metadata(experiment_accession_code)
+
+        r = requests.get(SAMPLES_URL.format(experiment_accession_code))
+        samples = r.json()["experiment"]["sample"]
+
+        batches = []
+        for sample in samples:
+            if "file" not in sample:
+                continue
+
+            organism_name = "UNKNOWN"
+            for characteristic in sample["characteristic"]:
+                if characteristic["category"].upper() == "ORGANISM":
+                    organism_name = characteristic["value"].upper()
+
+            if organism_name == "UNKNOWN":
+                logger.error("Sample from experiment %s did not specify the organism name.",
+                             experiment_accession_code)
+                organism_id = 0
+            else:
+                organism_id = Organism.get_id_for_name(organism_name)
+
+            for sample_file in sample["file"]:
+                if sample_file["type"] != "data" and sample_file["name"] is not None:
+                    continue
+
+                batches.append(Batch(
+                    size_in_bytes=-1,  # Will have to be determined later
+                    download_url=sample_file["comment"]["value"],
+                    raw_format=sample_file["name"].split(".")[-1],
+                    processed_format="PCL",
+                    platform_accession_code=experiment["platform_accession_code"],
+                    experiment_accession_code=experiment_accession_code,
+                    organism_id=organism_id,
+                    organism_name=organism_name,
+                    experiment_title=experiment["name"],
+                    release_date=experiment["release_date"],
+                    last_uploaded_date=experiment["last_update_date"],
+                    name=sample_file["name"]
+                ))
+
+        # Group batches based on their download URL and handle each group.
+        download_urls = set(map(lambda x: x.download_url, batches))
```

Comment:
I think @mhuyck had a similar suggestion in his review. I think you can do a set comprehension for this line to make it simpler:
``download_urls = {bt.download_url for bt in batches}``